### ISSUE TITLE
Tracing

### ISSUE THREAD

**Issue by danielepolencic** [uasabi/noderosso#72]
_TL;DR: create an open tracing plugin for node-red._

## tasks

- [ ] Implement the basic opentelemetry configuration for node.js and confirm that the it works with bazel.
- [ ] Include the node plugin for opentelemetry. It should track http calls by default.
- [ ] Make the tracing backend configurable. In development it should be the console, in production it should use honeycomb.
- [ ] the node-red plugin for opentelemetry should create a root span for every flow so that we can track the messages across nodes.
- [ ] The node-red plugin should include the message in the root span. For every node, it should include the properties but not the credentials. (not sure about this, though)

## notes

- node-red has hooks https://github.com/node-red/designs/pull/7#issuecomment-666516733

## links

- https://www.aspecto.io/blog/context-management-in-opentelemetry-node-js/
- https://opentelemetry.io/docs/js/getting_started/nodejs/

**Comment by danielepolencic** [comment]
Solved with https://github.com/uasabi/noderosso/pull/103
